wrangler-service/src/main/java/io/cdap/wrangler/service/directive/DirectivesHandler.java

```diff
@@ -663,6 +664,18 @@ private String addLoadablePragmaDirectives(String namespace, Request request) {
       // Compile the directive extracting the loadable plugins (a.k.a
       // Directives in this context).
       CompileStatus status = compiler.compile(new MigrateToV2(request.getRecipe().getDirectives()).migrate());
+      if (!status.isSuccess()) {
+        StringBuilder eStr = new StringBuilder();
+        Iterator<SyntaxError> errors = status.getErrors();
+        while (errors.hasNext()) {
```

Comment:
seems like this could potentially result in a giant error message that covers the whole screen. Should we limit the number of errors in the message?

Reply:
@albertshau this is compile error. So, I don't expect it to be very long as compiler will terminate on first error, but can have two to three lines max.

Reply:
If there are 10 syntax errors in the recipe, won't this result in 10 error messages concatenated together with commas? And if each individual message has a comma or period or multiple sentences, this can very easily become unreadable.

Reply:
No, we terminate on first one.

Reply:
then why bother iterating through all of these and concatenating instead of just getting the first one?